### PR TITLE
Updates for llvm3.9.1

### PR DESCRIPTION
**Pull request by sklam**
based on #199 

* Remove TargetData.add_pass
* Update buildscripts
* Replace LLVMGetDataLayout with LLVMGetDataLayoutStr
* Update debug IR test
* Enabled assertion in llvmdev build
* oProfile support remove (build fail if on)

### CODE REVIEW

**Comment by seibert** [comment]
@pitrou: Can you review?

File: conda-recipes/llvmdev/build.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@ CMAKE_COMMON_VARIABLES=" -DCMAKE_INSTALL_PREFIX=$PREFIX \
     -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD=host \
     -DLLVM_INCLUDE_TESTS=OFF -DLLVM_INCLUDE_UTILS=OFF \
     -DLLVM_INCLUDE_DOCS=OFF -DLLVM_INCLUDE_EXAMPLES=OFF \
-    -DLLVM_ENABLE_TERMINFO=OFF \
+    -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_ENABLE_ASSERTIONS=ON \
```
Comment (pitrou) [inline conda-recipes/llvmdev/build.sh]: This will be useful :-) I wonder at which point we dropped assertions? (probably by mistake)

**Comment by pitrou** [comment]
This looks good to me. I'm glad the migration is easier this time than it has been in the past...

**Comment by seibert** [comment]
Great, thanks everyone!

**Pull request merged**